perlock-core/src/main/java/name/mitterdorfer/perlock/impl/WatchServicePathWatcher.java

```diff
@@ -75,11 +83,13 @@ private void performRegistration() throws IOException {
      * Process all events for keys queued to the watcher
      */
     @Override
-    public void start() throws IOException {
+    public PathWatcher start() throws IOException {
         prepareWatcher();
         performRegistration();
         running = true;
+        lifecycleListener.onStart(this);
```

Comment:
Regarding 1, that is a good point that I missed, I will address it.

Regarding 2 I'm not sure exactly what the right approach would be, but I will take another look.

The important thing is that the user has a guaranteed way of knowing when the watch service has begun watching for events.
One thing I considered after submitting this PR was having the `RunnablePathWatcherAdapter#start()` method block until it actually starts.
I think it would be ideal to implement both, the accurate life-cycle and the start method blocking.